### ISSUE TITLE
Library -> Search

### ISSUE THREAD

**Issue by acolombier** [mixxxdj/mixxx#14534]
### Mockups

Recent search could show as set of token when using criteria search:
![Image](https://github.com/user-attachments/assets/54598448-7f07-41ae-accb-5de268817956)

Suggestions should include available criteria to search with:
![Image](https://github.com/user-attachments/assets/b2ecc22b-e07e-4a28-86a0-00a27c3fa4b9)

Everything should be possible to be done with the keyboard only. Note that we might want to consider "tap top accept" button to better support touchscreen/keyboard-less devices such as tablets
![Image](https://github.com/user-attachments/assets/6c7ada6e-d718-4f7c-9336-43a7a4ae0f0a)

![Image](https://github.com/user-attachments/assets/9af688fc-edc5-416c-aca9-db9f4c3e6828)

![Image](https://github.com/user-attachments/assets/bb3033b1-bf43-4a17-9751-1d6be0afa4fd)

![Image](https://github.com/user-attachments/assets/cb90fa7b-1f0d-40a8-a47d-2dd9fcda5640)

Active searches should show minified. There should be an easy way to dismiss the search or save it for later use 
![Image](https://github.com/user-attachments/assets/1b4bd1d4-098f-4521-81ac-1f8b2da5e9eb)
